### ISSUE TITLE
AudioKit incompatible with apps that use camera to record video

### ISSUE THREAD

**Issue by bennnjamin** [audiokit/AudioKit#875]
Hello, first of all thanks for the excellent project. I am trying to incorporate audio recording like the iOS voice memo app into my project which currently also has video recording using `UIImagePickerController`. The problem is after recording a video in my `videoViewController`, I switch back to the `audioViewController` and can no longer record audio or view the input plot. The audio engine freezes with no crash as far as I can tell.

I have replicated the issue using Recorder Demo project. In order to reproduce the issue you must perform these steps:

0) Grab iOS Recorder Demo Project from here https://github.com/bennnjamin/AudioKit/tree/master/Examples/iOS/RecorderDemo
1) Tap record video
2) Switch to Video in the camera preview
3) Press record button
4) Stop recording after a few seconds and click 'Use Video'
5) Tap on record audio

Now you will see the input plot does not work whatsoever

6) Tap Record to record an audio file
7) When done play back the audio and you will hear nothing, nothing is recorded

**Other Notes**

- Any modifications to `ViewController.swift` cause an AURemoteIO::IOThread EXC_BAD_ACCESS so I was not able to add a cancel button to the record audio view (without the app entirely crashing).
- You can get the audio to work again by force closing the app

**Comment by aure** [comment]
Huh, yeah, never tried using AudioKit with video.  Should this type of thing work with say just using AVAudioEngine?  If so, then it should work with AudioKit.  If not, then we might be out of luck.  I'm not sure there are any steps I should take to resolve this for you, but if you find something that helps you accomplish what you're trying to do, I'd love to know more about it.

**Comment by aure** [comment]
Also, its possible this is related to a bug fixed on the develop branch with a release coming soon.

**Comment by bennnjamin** [comment]
@aure it's possible it is an issue (i.e. bug) with the iOS platform. I found some sample code to record using AVAudioEngine directly and replaced AudioKit with that, and the issue is identical. You can record audio fine at first but then if you record a video and go back to audio, the file is filled with silence (still being written to though). 

Furthermore, if I used AVAudioRecorder, it works no problem! I am assuming AVAudioRecorder uses AVAudioEngine but since it's closed source it is possible Apple is using some other implementation internally.

**Comment by aure** [comment]
Yeah, it's very hard to tell what could be done about this, especially if AVAudioEngine (AudioKit's base) is exhibiting the same behavior.  Reopen if you figure out a workaround or whatever.

**Comment by ryanthon** [comment]
Hi,
I just want to chime in on this issue in case my findings lead to a solution...
I use AudioKit for a voice recording feature on my app. In the same workflow, I have a video recording option as well.

When I first built it out, when I presented the UIImagePickerController and switched to video mode, the app would crash if I had AudioKit running (with AudioKit.start()). I had to solve this by calling AudioKit.stop() before presenting the UIImagePickerController.

Once I solved that issue, what I see is that while the recording function worked, the output was either complete electronic noise/buzzing, or that it was a very garbled, noisy, electronic sounding version of what was actually recorded. I tried creating new AKMicrophone, AKMixer, etc. objects, but that doesn't work either. The sound doesn't go back to normal until the app is completely restarted.
